.github/workflows/test_lint_format.yml

```diff
@@ -0,0 +1,37 @@
+name: Test Lint and Format
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test-lint-format:
+    runs-on: ubuntu-latest
+
+    # dummy token for required bot env variable
+    env:
+      TOKEN: foo
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.12'
+
+      - name: Install uv and dependencies
+        run: |
+          pip install uv
+          uv venv
```

Comment:
do you really need a virtualenv in a virtualized GH runtime?

Reply:
lol yea i was thinking the same thing. I'm just so used to creating one. This is my first time writing a github workflow.

Reply:
undone. `uv` REALLY likes virtual environments, and I can't make it install the dependencies to the system globally while also splitting the dev and prod dependencies in pyproject.toml. I looked at all the commands and flags, and it just isn't possible.